src/moj/components/date-picker/date-picker.js

```diff
@@ -0,0 +1,596 @@
+  // eslint-disable-next-line prefer-destructuring
+  this.firstButtonInDialog = dialogButtons[0]
+  this.lastButtonInDialog = dialogButtons[dialogButtons.length - 1]
+  this.firstButtonInDialog.addEventListener('keydown', event => this.firstButtonKeyup(event))
+  this.lastButtonInDialog.addEventListener('keydown', event => this.lastButtonKeyup(event))
+
+  this.$calendarButton.addEventListener('click', event => this.toggleDialog(event))
+
+  document.body.addEventListener('mouseup', event => this.backgroundClick(event))
+
+  // populates calendar with initial dates, avoids Wave errors about null buttons
+  this.updateCalendar()
+}
+
+Datepicker.prototype.createDialogMarkup = function (titleId) {
+  return `<div class="moj-datepicker__dialog__header ">
+        <div class="moj-datepicker__dialog__navbuttons">
+            <button class="js-datepicker-prev-year" aria-label="previous year" data-button="button-datepicker-prevyear">
+                <span class="govuk-visually-hidden">Previous year</span>
+                <svg focusable="false" class="moj-datepicker-icon" aria-hidden="true" role="img"><use href="/assets/images/icons.stack.svg#double_chevron_left"></use></svg>
+            </button>
+
+            <button class="js-datepicker-prev-month" aria-label="previous month" data-button="button-datepicker-prevmonth">
+                <span class="govuk-visually-hidden">Previous month</span>
+                <svg focusable="false" class="moj-datepicker-icon" aria-hidden="true" role="img"><use href="/assets/images/icons.stack.svg#chevron_left"></use></svg>
+            </button>
+        </div>
+
+        <h2 id="${titleId}" class="moj-datepicker__dialog__title  js-datepicker-month-year" aria-live="polite">June 2020</h2>
+
+        <div class="moj-datepicker__dialog__navbuttons">
+            <button class="js-datepicker-next-month" aria-label="next month" data-button="button-datepicker-nextmonth">
+                <span class="govuk-visually-hidden">Next month</span>
+                <svg focusable="false" class="moj-datepicker-icon" aria-hidden="true" role="img"><use href="/assets/images/icons.stack.svg#chevron_right"></use></svg>
+            </button>
+
+            <button class="js-datepicker-next-year" aria-label="next year" data-button="button-datepicker-nextyear">
+                <span class="govuk-visually-hidden">Next year</span>
+                <svg focusable="false" class="moj-datepicker-icon" aria-hidden="true" role="img"><use href="/assets/images/icons.stack.svg#double_chevron_right"></use></svg>
+            </button>
+        </div>
+      </div>
+
+      <table class="moj-datepicker__dialog__table  js-datepicker-grid" role="grid" aria-labelledby="${titleId}">
+      <caption class="moj-datepicker__dialog__table-caption">You can use the arrow keys to select a date</caption>
+      <thead>
+          <tr>
+          <th scope="col" abbr="Monday">Mo</th>
```

Comment:
Should probably add `aria-label` to these to avoid screenreaders announcing "Mo"